testing/trino-testing-containers/src/main/java/io/trino/testing/containers/OpenTracingCollector.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing.containers;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static com.google.common.net.UrlEscapers.urlFragmentEscaper;
+
+public class OpenTracingCollector
+        extends BaseTestContainer
+{
+    private static final Path storageDirectory;
+
+    static {
+        try {
+            storageDirectory = Files.createTempDirectory("tracing-collector");
```

Comment:
class init shouldn't do IO. can this be perhaps lazy?

Reply:
https://github.com/trinodb/trino/pull/18307